src/test/java/org/opensearch/knn/index/memory/NativeMemoryCacheManagerTests.java

```diff
@@ -197,7 +197,7 @@ public void testGetTrainingSize() throws ExecutionException {
 
         nativeMemoryCacheManager.get(trainingDataEntryContext, true);
 
-        assertEquals((float) allocationEntryWeight, nativeMemoryCacheManager.getTrainingSizeInKilobytes(), 0.001);
+        assertEquals(allocationEntryWeight, nativeMemoryCacheManager.getTrainingSizeInKilobytes(), 1e-3);
```

Comment:
I fixed this flaky testing where it fails the testing by one.

```
(float) allocationEntryWeight == 2.1430328E7
nativeMemoryCacheManager.getTrainingSizeInKilobytes() == 21430329
```

Reply:
good catch